### DISCUSSION TITLE
Remove the flaw links when it can fallback to en-US in l10n

### DISCUSSION THREAD

**Discussion by awxiaoxian2020** [mdn/mdn-community#144] [Platform] [answered]
https://github.com/mdn/yari/blob/1cb12182f2aa524c1f9af4ec9eca4e238220f8b5/build/flaws/broken-links.js#L319

I think when we get a hint that there is a broken link. We should try to fix it. Otherwise, hint is less useful.

Use a unified localized link to facilitate maintenance. When there is a l10n of the content, jump to the l10n of the link, if not, jump to the English version. I think it's a good mechanism.
When the maintainers see that the localized link does not exist, they will not be able to add localized content in the short term. So I don't think this is a bug, just a good strategy that doesn't need hints.

**Comment by awxiaoxian2020** [comment]
Any response for this issue?

**Comment by bsmth** [comment]
Hi @awxiaoxian2020, thanks for opening the discussion. Pinging @mdn/core-dev :)

**Reply by caugner** [reply]
@mdn/localization-team-leads The OP suggests to remove the broken link flaw for cases where we can fallback to `en-US`. Does this make sense to you?

**Reply by cw118** [reply]
I personally think this makes sense. I might be wrong, but the fallback to `en-US` link flaw seems to encourage pointing links to the `en-US` page if its translation does not exist. This can make maintaining translated content more difficult if we have to replace these `en-US` links once the page is localized, especially when the links already automatically fall back to the English page and will point to the localized version when it is created, as expected. 

Unless there are other uses for the flaw, I do not think it is needed.

**Reply by mfuji09** [reply]
I agree to remove reporting link flaws if they can be redirected to `en-US` pages. Because:

1. They are not flaws in the reported page.
2. They shouldn't be fixed not to be reported the flaw, as @cw118 said.
3. Also, if the pages linked to from the sidebar have not yet been translated, many flaws are reported that cannot be fixed immediately, covering up the flaws that should be fixed,
  For example, these flaws reported are due to lack localized pages linked in the sidebar:
  ![image](https://github.com/mdn/mdn-community/assets/11516302/965a2531-dcc0-419d-b939-491fd6943791)

The pages which haven't been translated yet are easily can be get, for example:
- getting diff between the language's and `en-US` contents
- via the localization dashboard (http://localhost:3000/xx/_translations/dashboard) or missing page.

**Comment by awxiaoxian2020** [comment]
So would anyone like to submit a PR to move this forward?

**Reply by yin1999** [reply, marked as answer]
I've created a PR for this: mdn/yari#9408. But we may also need to reorganize the repair logic for URL.

**Comment by yin1999** [comment]
Hey all. I think @awxiaoxian2020 created this discussion to address the flaw reporting in `broken-links.js`, and this has been resolved in mdn/yari#9408. While the screenshot provided by @mfuji09 shows that there are many missing localized page flaws (`MacroBrokenLinkError` caused by [this](https://github.com/mdn/yari/blob/e8ec3d3b888950cc4a3d052241f93216a5b5d9f5/kumascript/src/api/web.ts#L142-L153)) reported by the kuma `macro`s. I think we should also remove the flaws reporting related to it. Let me know your thought :)
